stdlib/fun.mli

```diff
@@ -31,3 +31,29 @@ val flip : ('a -> 'b -> 'c) -> ('b -> 'a -> 'c)
 val negate : ('a -> bool) -> ('a -> bool)
 (** [negate p] is the negation of the predicate function [p]. For any
     argument [x], [(negate p) x] is [not (p x)]. *)
+
+val protect : acquire:(unit -> 'a) -> release:('a -> unit) -> ('a -> 'b) -> 'b
+(** [protect ~acquire ~release work] invokes [acquires], then [work],
+   and then invokes [release] when [work] returns with its value or an
+   exception. In the latter case the exception is re-raised after
+   [release]. If [release] raises an exception, then the exception
+   [Release_failure] is raised instead (see [Release_failure] below).
```

Comment:
I think you can remove the parenthesis and simply turn [Release_failure] into a link (`{!Release_failure}`).